CMakeLists.txt

```diff
@@ -293,7 +293,7 @@ if(CONFIG_COMPILER_FREESTANDING)
   zephyr_compile_options($<$<COMPILE_LANGUAGE:ASM>:$<TARGET_PROPERTY:compiler,freestanding>>)
 endif()
 
-if (CONFIG_PICOLIBC AND NOT CONFIG_PICOLIBC_IO_FLOAT)
+if (CONFIG_PICOLIBC AND NOT CONFIG_PICOLIBC_IO_FLOAT AND NOT DEFINED ZEPHYR_SCA_VARIANT)
```

Comment:
Is there a mechanism to detect `clang` instead of looking at `ZEPHYR_SCA_VARIANT`?
The issue #62278 is specific for `clangsa` and `clang-tidy`, others can be compatible with `gcc` arguments.

Reply:
Yeah, I'm actually wondering if this could be fixed in CodeChecker instead of Zephyr. That tool might already have a way to adjust the compiler flags for this kind of issue? 

This particular compiler flag is necessary to avoid some gcc optimizations which compute the return value from the sprintf family of functions at compile time, so we get incorrect code unless sprintf is fully c11-conforming (which requires `PICOLIBC_IO_FLOAT`). That means that with `ZEPHYR_SCA_VARIANT` defined, we'll be generating broken code here.

Reply:
I'd love to rework the `CodeChecker` SCA plugin to use the logger feature, instead of just using the `compile_commands.json` file, see https://github.com/Ericsson/codechecker/tree/master/analyzer/tools/build-logger
But I lack some compiler/toolchain experience to make it work.